pkg/gardenlet/controller/managedseed/charttest/charttest.go

```diff
@@ -1015,64 +1015,96 @@ func ComputeExpectedGardenletDeploymentSpec(
 
 		if deploymentConfiguration.ReplicaCount != nil {
 			deployment.Replicas = deploymentConfiguration.ReplicaCount
-			deployment.Template.Spec.Affinity = &corev1.Affinity{
-				PodAntiAffinity: &corev1.PodAntiAffinity{
-					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
-						{
-							LabelSelector: &metav1.LabelSelector{
-								MatchExpressions: []metav1.LabelSelectorRequirement{
-									{
-										Key:      "app",
-										Operator: "In",
-										Values:   []string{"gardener"},
-									},
-									{
-										Key:      "role",
-										Operator: "In",
-										Values:   []string{"gardenlet"},
-									},
-								},
-							},
-							TopologyKey: "kubernetes.io/hostname",
+		}
+
+		topologySpreadConstraintLabels := map[string]string{
+			"app":      "gardener",
+			"role":     "gardenlet",
+			"chart":    "runtime-0.1.0",
+			"release":  "gardenlet",
+			"heritage": "Tiller",
+		}
+
+		if failureToleranceType := deploymentConfiguration.FailureToleranceType; failureToleranceType != nil {
```

Comment:
Shouldn't this be executed only if the `deployment.Replicas > 1`? From the `_helpers.tpl` I see that the `gardenlet.deployment.topologySpreadConstraints` only has content if `.Values.global.gardenlet.replicaCount` is greater than 1. Adding a test entry that looks like this under https://github.com/gardener/gardener/blob/9a0b8d5375f49dd120db559df928e0236f2ae366/pkg/gardenlet/controller/managedseed/charttest/gardenlet_chart_test.go#L399-L402:
```
Entry("verify deployment with replica count 1 and `zone` failureTolerance", nil, nil, nil, nil, nil, nil, &seedmanagement.GardenletDeployment{
			ReplicaCount:         pointer.Int32(1),
			FailureToleranceType: &failureToleranceZone,
		}, nil, nil, nil, map[string]string{"gardenlet-configmap": "gardenlet-configmap-a99e2d4b"}),
```
returns an error.